src/videojs.ads.js

```diff
@@ -617,19 +617,16 @@ var
             if (!videojs.browser.IS_IOS && player.duration() === Infinity) {
               player.volume(this.preAdVolume_);
             }
-
-            // trigger 'adend' as a consistent notification
-            // event that we're exiting ad-playback.
-            if (player.ads.triggerevent !== 'adend') {
-              player.trigger('adend');
```

Comment:
Yes, that is why we sometimes have double `adend` events.
We just want to make sure we notify with an `adend` every time we leave `ad-playback`. The only other time that happens is when `adserror` is triggered, thus making sense to move trigger('adend')